FirebaseSegmentation/Sources/SEGContentManager.m

```diff
@@ -63,29 +63,28 @@ - (instancetype)initWithDatabaseManager:databaseManager networkManager:networkMa
   return self;
 }
 
-// TODO(dmandar) IID only supports default instance. Modify for FIS.
-- (FIRInstanceID *)instanceIDForApp:(NSString *)firebaseApp {
-  return [FIRInstanceID instanceID];
+- (FIRInstallations *)installationForApp:(NSString *)firebaseApp {
+  return [FIRInstallations installationsWithApp:[FIRApp appNamed: firebaseApp]];
 }
 
 - (void)associateCustomInstallationIdentiferNamed:(NSString *)customInstallationID
                                       firebaseApp:(NSString *)firebaseApp
                                        completion:(SEGRequestCompletion)completionHandler {
-  // Get the latest instance identifier
-  if (![self instanceIDForApp:firebaseApp]) {
-    completionHandler(NO, @{kErrorDescription : @"InstanceID SDK not available"});
+  // Get the latest installation identifier
+  if (![self installationForApp:firebaseApp]) {
+    completionHandler(NO, @{kErrorDescription : @"Firebase Installations SDK not available"});
   }
   __weak SEGContentManager *weakSelf = self;
-  [[FIRInstanceID instanceID] instanceIDWithHandler:^(FIRInstanceIDResult *_Nullable result,
-                                                      NSError *_Nullable error) {
+  [[self installationForApp:firebaseApp] installationIDWithCompletion:^(NSString *__nullable identifier,
```

Comment:
Using `self` here defeats the purpose of capturing `weakSelf`. `strongSelf` should be used in the block. To help catching such errors you may just move the code in the block to a separate method, so you have to call `strongSelf` only once and use regular `self` in the method. But it will be a bigger refactoring, so may be done separately.

Reply:
Thank you maksym. I've noted down the issue. We'll do it in a separate PR.

Reply:
done. Thank you!